components/host-dashboard/reports-section/HostFeesSection.js

```diff
@@ -0,0 +1,217 @@
+      height: 16,
+    },
+  },
+  dataLabels: { enabled: false },
+  grid: {
+    raw: { opacity: 0 },
+    column: { opacity: 0 },
+    xaxis: { lines: { show: true } },
+    yaxis: { lines: { show: false } },
+  },
+  plotOptions: {
+    bar: {
+      columnWidth: '50%',
+    },
+  },
+  colors: ['#46347F', '#95DDF4', '#F5C451', '#0EA755'], // TODO(HostReport): Use host primary colors
+  xaxis: {
+    categories: [...new Array(12)].map(
+      (_, idx) => `${intl.formatDate(new Date(0, idx), { month: 'short' }).toUpperCase()}`,
+    ),
+  },
+  yaxis: {
+    labels: {
+      minWidth: 40,
+      formatter: function (value) {
+        if (value >= 1000) {
+          return `${Math.round(value / 1000)}k`;
+        } else {
+          return formatCurrency(value, hostCurrency);
+        }
+      },
+    },
+  },
+});
+
+const getHostFeesWithoutShare = (hostFeeNodes, hostFeeShareNodes) => {
+  const totalHostFeeSharePerMonthInCents = hostFeeShareNodes.reduce((result, node) => {
+    const monthKey = new Date(node.date).getMonth();
+    result[monthKey] = (result[monthKey] || 0) + node.amount.valueInCents;
+    return result;
+  }, {});
+
+  return hostFeeNodes.map(node => {
+    const monthKey = new Date(node.date).getMonth();
+    if (totalHostFeeSharePerMonthInCents[monthKey]) {
+      const valueInCents = node.amount.valueInCents - totalHostFeeSharePerMonthInCents[monthKey];
+      return { ...node, amount: { ...node.amount, valueInCents, value: valueInCents / 100 } };
+    } else {
+      return node;
+    }
+  });
+};
+
+const getSeriesFromData = (intl, timeSeries) => {
+  const dataToSeries = data => {
+    const series = new Array(12).fill(0); // = 12 months
+    data?.forEach(({ date, amount }) => (series[new Date(date).getMonth()] = amount.value));
+    return series;
+  };
+
+  const hostFeeNodes = get(timeSeries, 'hostFees.nodes', []);
+  const hostFeeShareNodes = get(timeSeries, 'hostFeeShare.nodes', []);
+  return [
+    { name: 'Host profit', data: dataToSeries(getHostFeesWithoutShare(hostFeeNodes, hostFeeShareNodes)) },
+    ...Object.entries(groupBy(hostFeeShareNodes, 'settlementStatus')).map(([status, nodes]) => ({
+      name: `Host fee share (${i18nTransactionSettlementStatus(intl, status)})`,
+      data: dataToSeries(nodes),
+    })),
+  ];
+};
+
+const getActiveYearsOptions = host => {
+  const currentYear = new Date().getFullYear();
+  const firstYear = host ? parseInt(host.createdAt.split('-')[0]) : currentYear;
+  const activeYears = [...Array(currentYear - firstYear + 1).keys()].map(year => year + firstYear);
+  return activeYears.map(year => ({ value: year, label: year }));
+};
+
+const getQueryVariables = (hostSlug, year) => {
+  return {
+    hostSlug,
+    startDate: `${year}-01-01T00:00:00Z`,
+    endDate: `${year}-12-31T23:59:59Z`,
+  };
+};
+
+const HostFeesSection = ({ hostSlug }) => {
+  const intl = useIntl();
+  const [selectedYear, setSelectedYear] = React.useState(() => new Date().getFullYear());
```

Comment:
```suggestion
  const [selectedYear, setSelectedYear] = React.useState(new Date().getFullYear());
```

Don't think we need to have a function here? 🤔

Reply:
The main advantage of using a function is that it will only run on the first render. So if you do:

```es6
React.useState(() => expensiveTask());
```

`expensiveTask` will only run once.

Here it's value is debatable, as `new Date().getFullYear()` is not an heavy process. In my opinion, it's a cheap optimization that doesn't really impact readability.

Reply:
Ah, that makes sense. Thanks for the explanation.  👍🏽